llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -851,6 +863,9 @@ void VPlan::execute(VPTransformState *State) {
   State->CFG.ExitBB = State->CFG.PrevBB->getSingleSuccessor();
   BasicBlock *VectorPreHeader = State->CFG.PrevBB;
   State->Builder.SetInsertPoint(VectorPreHeader->getTerminator());
+  replaceVPBBWithIRVPBB(
```

Comment:
Another option may be to place this in prepareToExecute(), as its finalizing VPlan before actually generating code.